test/test_time.cpp

```diff
@@ -117,3 +118,109 @@ TEST_F(TestTimeFixture, test_rcutils_steady_time_now) {
   EXPECT_LE(
     llabs(steady_diff - sc_diff), RCUTILS_MS_TO_NS(k_tolerance_ms)) << "steady_clock differs";
 }
+
+// Tests the rcutils_time_point_value_as_nanoseconds_string() function.
+TEST_F(TestTimeFixture, test_rcutils_time_point_value_as_nanoseconds_string) {
+  rcutils_ret_t ret;
+  rcutils_time_point_value_t timepoint;
+  char buffer[256] = "";
+
+  // Typical use case.
+  timepoint = 100;
+  ret = rcutils_time_point_value_as_nanoseconds_string(&timepoint, buffer, sizeof(buffer));
+  EXPECT_EQ(RCUTILS_RET_OK, ret) << rcutils_get_error_string_safe();
+  EXPECT_STREQ("0000000000000000100", buffer);
+
+  // nullptr for timepoint
+  ret = rcutils_time_point_value_as_nanoseconds_string(nullptr, buffer, sizeof(buffer));
+  EXPECT_EQ(RCUTILS_RET_INVALID_ARGUMENT, ret);
+  rcutils_reset_error();
+
+  // nullptr for string
+  timepoint = 100;
+  ret = rcutils_time_point_value_as_nanoseconds_string(&timepoint, nullptr, 0);
+  EXPECT_EQ(RCUTILS_RET_INVALID_ARGUMENT, ret);
+  rcutils_reset_error();
+
+  // test truncations
+  timepoint = 100;
+  ret = rcutils_time_point_value_as_nanoseconds_string(&timepoint, buffer, 18);
+  EXPECT_EQ(RCUTILS_RET_OK, ret) << rcutils_get_error_string_safe();
+  EXPECT_STREQ("00000000000000001", buffer);
+
+  const char * test_str = "should not be touched";
+  timepoint = 100;
+  (void)memmove(buffer, test_str, std::min(sizeof(buffer), strlen(test_str)));
```

Comment:
Correct me if I'm wrong, but if we're gonna go the `std::min` route, then:
If `std::min` returns `sizeof(buffer)`, shouldn't we manually null terminate `buffer`?
In the other case, I think you'll want to copy `strlen(test_str)+1` (or manually null terminate `buffer`).

Reply:
You're right, I'll refactor this and add an expect case which checks for properly null terminated strings.